site/cds_rdm/ldap/user_importer.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-RDM invenio user importer API."""
+
+from flask import current_app
+from invenio_accounts.models import User
+from invenio_db import db
+from invenio_oauthclient.models import RemoteAccount, UserIdentity
+from invenio_userprofiles.models import UserProfile
+
+
+class LdapUserImporter:
+    """Import ldap users to Invenio RDM records.
+
+    Expected input format for ldap users:
+        [
+            {'displayName': [b'Joe Foe'],
+             'department': [b'IT/CDA'],
+             'uidNumber': [b'100000'],
+             'mail': [b'joe.foe@cern.ch'],
+             'cernAccountType': [b'Primary'],
+             'employeeID': [b'101010']
+            },...
+        ]
+    """
+
+    def __init__(self):
+        """Constructor."""
+        self.client_id = current_app.config["CERN_APP_CREDENTIALS"]["consumer_key"]
+
+    def create_invenio_user(self, ldap_user):
+        """Commit new user in db."""
+        email = ldap_user["user_email"]
+        username = ldap_user["keycloak_id"]
+        user = User(email=email, username=username, active=True)
```

Comment:
Here we set the username to the keycloak_id/ldpa cn, just want to double check that it's okay to expose this as the username

Reply:
it is already exposed in the webservices internally if I am not mistaken, also via e-groups